pkg/clients/database/mysql.go

```diff
@@ -47,6 +50,10 @@ const (
 	StateDisabled = string(mysql.ServerStateDisabled)
 	StateDropping = string(mysql.ServerStateDropping)
 	StateReady    = string(mysql.ServerStateReady)
+
+	AsyncOperationStatusInProgress = "InProgress"
```

Comment:
What would you think about this (and potentially the state constants) living in the api pkg as opposed to the client pkg?

Reply:
This one could live in the `api` package but not the state constants unfortunately because they are taken from `mysql` package of Azure. Including them in `api` package would require consumers of the `api` package to have dependency on Azure.

On the other hand, though it'd require a code change when their values change, it makes sense to just get the value as bare string from the library and put in the `api` package. It's very rare that the state strings would change. They could be useful for the type's consumers as they'd want to see the possible states. I'd like to do that in a separate PR.

Reply:
Separate PR sounds good to me :)

Reply:
Nevermind, I just pushed another commit for it :)